app/PostComponent.tsx

```diff
@@ -0,0 +1,105 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { FiHeart as HeartIcon } from 'react-icons/fi';
+import { GoPaperAirplane as ShareIcon } from 'react-icons/go';
+import { LuCircle as ProfileIcon } from 'react-icons/lu';
+import { supabase } from '@/supabase/client';
+import styles from './styles.module.css';
+import '../assets/global.css';
+
+interface Post {
+  author: string;
+  date: string;
+  image_url: string;
+  content: string;
+  likes: number;
+}
+
+export default function PostComponent() {
+  // Define a state variable to store the fetched data
+  const [postData, setPostData] = useState<Post[]>([]);
+
+  async function fetchPostData() {
+    // TODO: Fetch data using the supabase client
+    const response = await supabase.from('posts').select('*');
+    const { data, error } = response;
+    if (error) {
+      throw error;
+    }
+    const formattedData: Post[] = data.map(post => ({
+      author: post.author,
+      date: post.date,
+      image_url: post.image_url,
+      content: post.content,
+      likes: post.likes,
+    }));
+
+    setPostData(formattedData);
+  }
+
+  // useEffect with an empty dependency array runs only once after the component is first rendered
+  useEffect(() => {
+    // Inside useEffect, you can perform side effects, like fetching data
+    // Here, we fetch some data from an API
+    fetchPostData();
+  }, []); // Empty dependency array means this effect runs only once
+
+  // Render the fetched data
+  return (
+    <div>
+      {postData == null ? (
+        <p>Loading...</p>
+      ) : (
+        postData.map(post => (
+          <main key={post.date} className={styles.main}>
```

Comment:
You should make this a Component! You can make a separate file with this code, add props (which will be similar to the post interface, and render it here. It'll make the code more readable.